services/download/src/main/java/app/coronawarn/server/services/download/config/DownloadServiceConfigValidator.java

```diff
@@ -0,0 +1,49 @@
+
+
+package app.coronawarn.server.services.download.config;
+
+import java.util.Map;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+/**
+ * Validate the values of the DownloadServiceConfig.
+ */
+public class DownloadServiceConfigValidator implements Validator {
+
+  @Override
+  public boolean supports(Class<?> type) {
+    return type == DownloadServiceConfig.class;
+  }
+
+  @Override
+  public void validate(Object configurationObject, Errors errors) {
+    DownloadServiceConfig properties = (DownloadServiceConfig) configurationObject;
+    validateTransmissionRiskLevelDerivationMap(errors, properties);
+  }
+
+  private void validateTransmissionRiskLevelDerivationMap(Errors errors, DownloadServiceConfig properties) {
+    Map<Integer, Integer> transmissionRiskLevelFromDaysSinceOnsetOfSymptoms =
+        properties.getTekFieldDerivations().getTransmissionRiskLevelFromDaysSinceOnsetOfSymptoms();
+
+    transmissionRiskLevelFromDaysSinceOnsetOfSymptoms.forEach((daysSinceOnsetOfSymptoms, transmissionRiskLevel) -> {
+      checkTransmissionRiskLevelInAllowedRange(transmissionRiskLevel, errors);
+      checkDaysSinceOnsetOfSymptomsInAllowedRange(daysSinceOnsetOfSymptoms, errors);
+    });
+  }
+
+  private void checkTransmissionRiskLevelInAllowedRange(Integer transmissionRiskLevel, Errors errors) {
+    if (transmissionRiskLevel > 8 || transmissionRiskLevel < 1) {
+      errors.rejectValue("tekFieldDerivations",
+          "[" + transmissionRiskLevel + "]: transmissionRiskLevel value is not in the allowed range (1 to 8)");
+    }
+  }
+
+  private void checkDaysSinceOnsetOfSymptomsInAllowedRange(Integer daysSinceOnsetOfSymptoms, Errors errors) {
+    if (daysSinceOnsetOfSymptoms > 4000 || daysSinceOnsetOfSymptoms < -14) {
```

Comment:
Checking for valid DSOS values is also done on [key level here](https://github.com/corona-warn-app/cwa-server/blob/master/services/download/src/main/java/app/coronawarn/server/services/download/validation/ValidFederationKeyFilter.java#L27-L35).

I wonder if we should have this range `-14, 4000` as a constant somewhere, like we for [specific ENF constants](https://github.com/corona-warn-app/cwa-server/blob/master/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/validation/ParameterSpec.java).

Reply:
Yes, we will need to either externalize these rangs somewhere (both TRL and DSOS) or centralize them in a constants class under a common module.  This needs ot be done for submission as well, but we are waiting for some clarifications regarding the actual values of dsos.

Reply:
Fair enough. I don't think this stops this PR, just wanted to raise this.

Reply:
Yes, agree with you here. This should be improved. Maybe even take this a step further an centralize the mapping between the two values completely not only the allowed ranges? For the Trl to Dsos mapping we already have two places to maintain it, if something changes.